Analise_horas.py

```diff
@@ -0,0 +1,126 @@
+import pandas as pd
+import datetime
+import numpy as np
+
+
+WORKDAY = datetime.timedelta(hours=9, minutes=0, seconds=0)
+# 8 hours of work +1 lunch time
+WORKDAY_WEEKAND = datetime.timedelta(hours=4)
+# 4 hours of work on weekend
+
+
+def data_anal():
+    data = pd.read_csv("sent_emails.csv", ";")
+    df_data = pd.DataFrame()
+
+    df_data._reindex_columns = ["Date", "Week_day", "Worked_hours", "Overtime"]
+
+    print(data)
+    data["Date"] = data.Date.str.split(".", expand=True)
+    data["Date"] = data.Date.str.split("+", expand=True)
+    data[["Date", "Hour"]] = data.Date.str.split(" ", expand=True)
+
+    data["Date"] = pd.to_datetime(data["Date"], format="%Y-%m-%d")
+
+    def make_timedelta(time):
+        print(time)
+        hour, minute, seconds = time.split(":")
+        hour, minute, seconds = int(hour), int(minute), int(seconds)
+        hora = datetime.timedelta(hours=hour, minutes=minute, seconds=seconds)
+        return hora
+
+    data["Hour"] = data["Hour"].apply(make_timedelta)
+    # converts string to time delta
+
+    hour_bound = datetime.timedelta(hours=8)
+    filt_hour = data["Hour"] >= hour_bound
+    data = data[filt_hour]
+    # removes all e-mails sent between as 24 an 8 am.
+
+    df_data["Date"] = data["Date"].drop_duplicates()
+
+    df_data["Week_day"] = df_data["Date"].dt.dayofweek
+
+    df_data = df_data.reset_index(drop=True)
+
+    inicial = []
+    final = []
+    number_mails = []
+    for date in df_data["Date"]:
+        hour = data.loc[data["Date"] == date, "Hour"]
+        inicial.append(hour.min())
+        final.append(hour.max())
+        number_mails.append(len(hour))
+
+    df_data.insert(2, column="Last_mail", value=final, allow_duplicates=False)
+    # Column with hour of last sent e-mail
+    df_data.insert(2, column="First_mail", value=inicial, allow_duplicates=False)
+    # Column with first sent e-mail
+    df_data["Worked_hours"] = df_data["Last_mail"] - df_data["First_mail"]
+    # Column with worked hours
+    df_data.insert(5, column="Num_Mails", value=number_mails, allow_duplicates=False)
+    # Column with the number of mails sent
+
+    filt_sat = df_data["Week_day"] == 5
+    filt_sun = df_data["Week_day"] == 6
+
+    df_results_fds = df_data.loc[filt_sat | filt_sun]
+
+    df_results_sem = df_data.loc[~filt_sat | ~filt_sun]
```

Comment:
Nao sei se isto esta' correcto. Deduzo que queiras filtrar o que nao e' dia da semana, logo o que nao seja sabado e domingo. Mas o que isto esta a fazer aqui e' a dizer o que nao seja sabado ou nao seja domingo.

Se estou a ler bem , o que querias aqui era o oposto de `filt_sat || filt_sun`.

Gonna drop some computer science on you. Em logica, quando queres converter um predicado para o seu oposto, podes usar a [De-Morgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). Basicamente, se queres negar uma proposicao `A ^ B`, tens que negar os predicados e a operacao. Ou seja, `~(A ^ B) == ~A v ~B`.

TLDR: Se as minhas suposicoes estao correctas,
```suggestion
    df_results_sem = df_data.loc[~filt_sat & ~filt_sun]
```

Reply:
As tuas suposições estão correctas. Falta de raciocínio meu. Nem sequer é erro de programação... é raciocínio lógico. :(
Corrigido

Reply:
E' um erro muito facil de fazer. Se eu nao soubesse a lei De Morgan's tropecava nisto varias vezes, mas quando aprendes o algoritmo da lei ja nao te apanhas nesta! E' daquelas coisas raras que se aprende na universidade que sao practicas.

Reply:
Sim sim... eu também aprendi esta merda. Em analise matematica, no primeiro ano de universidade aprendi muito de logica e este tipo de leis.... mas já foi há mais de 10 anos!! Tenho de ir buscar os meus apontamentos e seriamente rever os temas. Acho que me vai fazer muito bem a nivel de programação.